adaptive/learner/balancing_learner.py

```diff
@@ -113,50 +113,67 @@ def strategy(self, strategy):
                 ' strategy="npoints" is implemented.')
 
     def _ask_and_tell_based_on_loss_improvements(self, n):
-        points = []
-        loss_improvements = []
+        selected = [] # tuples ((learner_index, point), loss_improvement)
+        npoints = [l.npoints + len(l.pending_points) for l in self.learners]
```

Comment:
Maybe `total_points` (to indicate that it's evaluated and pending together)?

Reply:
Done.